imcsdk/apis/server/serveractions.py

```diff
@@ -46,22 +49,42 @@ def power_up_server(handle):
         power_up_server(handle)
     """
 
+    import time
     from imcsdk.mometa.compute.ComputeRackUnit import ComputeRackUnit,\
         ComputeRackUnitConsts
 
-    rack_mo = ComputeRackUnit(parent_mo_or_dn="sys", server_id="1")
-    rack_mo.admin_power = ComputeRackUnitConsts.ADMIN_POWER_UP
-    handle.set_mo(rack_mo)
-    return rack_mo
-
-
-def power_down_server(handle):
+    # Turn power on only if not already powered up
+    if get_server_power_state(handle) != "on":
+        rack_mo = ComputeRackUnit(parent_mo_or_dn="sys", server_id="1")
+        rack_mo.admin_power = ComputeRackUnitConsts.ADMIN_POWER_UP
+        handle.set_mo(rack_mo)
+
+    # Poll until the server is powered up
+    wait_time = 0
+    while get_server_power_state(handle) != "on":
+        # Raise error if we've reached timeout
+        if wait_time > timeout:
```

Comment:
@britthouser For accuracy this should be `wait_time >= timeout`, otherwise we would end up waiting for another $interval duration because of the sleep.